pkg/controller/sesv2/tags.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sesv2
+
+import (
+	"sort"
+
+	svcsdk "github.com/aws/aws-sdk-go/service/sesv2"
+	"github.com/aws/aws-sdk-go/service/sesv2/sesv2iface"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	"github.com/pkg/errors"
+
+	svcapitypes "github.com/crossplane/provider-aws/apis/sesv2/v1alpha1"
+	awsclient "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+const (
+	errListTagsForResource = "cannot list tags"
+	errRemoveTags          = "cannot remove tags"
+	errCreateTags          = "cannot create tags"
+)
+
+// AreTagsUpToDate for spec and resourceName
+func AreTagsUpToDate(spec []*svcapitypes.Tag, current []*svcsdk.Tag) (bool, error) {
+	add, remove := DiffTags(spec, current)
+
+	return len(add) == 0 && len(remove) == 0, nil
+}
+
+// UpdateTagsForResource with resourceName
+func UpdateTagsForResource(client sesv2iface.SESV2API, spec []*svcapitypes.Tag, resourceArn *string) error {
```

Comment:
Seems like this is never used.

Reply:
Yes, keeping the code once able to figure out: // TODO(kelvinwijaya): Updating Tags will be useful if we can determine the resourceArn

Reply:
Whats blocking this TODO from being implemented?

Reply:
As for ConfigurationSet resource, the update of Tags is passed via CreateConfigurationSet API only: https://docs.aws.amazon.com/ses/latest/APIReference-V2/API_CreateConfigurationSet.html

As for today, the tagging API doesn't support sub resources tagging (e.g. configuratinset) and the only resource that can use this is emailidentity which has the arn based resource name: https://docs.aws.amazon.com/ses/latest/APIReference-V2/API_TagResource.html

I am keeping the code for future usage, but i can remove if this cause confusion as not being used for now.